empirical-data/common/src/main/scala/com/azavea/ingest/common/HydrateRDD.scala

```diff
@@ -62,15 +62,20 @@ object HydrateRDD {
     builder.buildFeature(orig.getID)
   }
 
-  def normalizeShpRdd(rdd: RDD[SimpleFeature], typeName: String) = {
-    val origSFT = rdd.first.getType
-    val builder = new SimpleFeatureTypeBuilder
-    builder.setName(typeName)
-    builder.addAll(origSFT.getAttributeDescriptors)
-    val sft = builder.buildFeatureType
+  def normalizeShpRdd(rdd: RDD[SimpleFeature], typeName: String)(implicit sc: SparkContext) =
+    rdd.mapPartitions({ featureIter =>
+      val bufferedFeatures = featureIter.buffered
+      val headFeature = bufferedFeatures.head
 
-    rdd.map(convertToSFT(sft))
-  }
+      val builder = new SimpleFeatureTypeBuilder
+      builder.addAll(headFeature.getType.getAttributeDescriptors)
+      builder.setName(typeName)
```

Comment:
This should just change the type - if there's user data on these features, I think it should remain. Perhaps there's a gap in my reasoning here